### PR TITLE
Router (and index.js) for React app.

### PR DESCRIPTION
**Pull request by nworden**

### CODE REVIEW

File: ui/src/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
```
Comment (gimite) [inline ui/src/index.js]: Be consistent whether to use single quotes or double quotes.

File: ui/src/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import {addLocaleData, IntlProvider} from 'react-intl';
+
+import App from "./App.js";
+
+addLocaleData(require('react-intl/locale-data/' + BUNDLE_LANGUAGE));
+
+ReactDOM.render(
+    <IntlProvider
+        locale={BUNDLE_LANGUAGE}
+        messages={require('./translations/' + BUNDLE_LANGUAGE + '.js')}>
```
Comment (gimite) [inline ui/src/index.js]: Optional: This may look better:

```js
`./translations/${BUNDLE_LANGUAGE}.js`
```

File: ui/src/App.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import RepoHome from "./pages/RepoHome.js";
+import Results from "./pages/Results.js";
+import View from "./pages/View.js";
+
+import './css/all.scss';
+
+const App = () => (
+  <BrowserRouter>
+    <div>
+      <Route exact path="/" component={GlobalHome} />
```
Comment (gimite) [inline ui/src/App.js]: You don't need to support those initially, but FYI. May be worth leaving TODO. We should support these URLs for backward compatibility:

* /global/home.html (had been the canonical URL of the global home page for long)
* /:repoId/ (with a trailing slash)

File: ui/src/index.js (hunk 1 above)
Comment (nworden) [reply ui/src/index.js]: The Google JS guide prefers single so I went with that.

File: ui/src/index.js (hunk 2 above)
Comment (nworden) [reply ui/src/index.js]: Done.

File: ui/src/App.js (hunk 3 above)
Comment (nworden) [reply ui/src/App.js]: Added a todo for global/home.html. Trailing slashes already work; I guess React Router just handles that for you.

File: ui/src/App.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
```
Comment (gimite) [inline ui/src/App.js]: Change double quotes in this file into single quotes as well.

File: ui/src/index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {addLocaleData, IntlProvider} from 'react-intl';
+
+import App from './App.js';
+
+addLocaleData(require('react-intl/locale-data/' + BUNDLE_LANGUAGE));
```
Comment (gimite) [inline ui/src/index.js]: Optional: You can write
```js 
    `react-intl/locale-data/${BUNDLE_LANGUAGE}`
```
here too which may look more consistent with one below.

File: ui/src/App.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import RepoHome from "./pages/RepoHome.js";
+import Results from "./pages/Results.js";
+import View from "./pages/View.js";
+
+import './css/all.scss';
+
+const App = () => (
+  <BrowserRouter>
+    <div>
+      /**
```
Comment (gimite) [inline ui/src/App.js]: Use a normal comment // ... instead of JSDoc /** ... */ for this.

File: ui/src/App.js (hunk 4 above)
Comment (nworden) [reply ui/src/App.js]: Done.

File: ui/src/App.js (hunk 6 above)
Comment (nworden) [reply ui/src/App.js]: Done.

File: ui/src/index.js (hunk 5 above)
Comment (nworden) [reply ui/src/index.js]: Done.

**Pull request merged**